### DISCUSSION TITLE
Azure DevOps - backup CosmosDB

### DISCUSSION THREAD

**Discussion by krukowskid** [AzureCommunityPL/group#38] [Pomoc]
Cześć, jaki jest najlepszy sposób na zrobienie kopii zapasowej bazy cosmosdb  (najlepiej do blob'a). Obecnie robię to ręcznie przed każdym wdrożeniem. Wykorzystując narzędzie "dt" robię kopię do .json i wrzucam do blob, Znalazłem taki poradnik, w którym wykorzystywany jest data factory https://ballardchalmers.com/2018/12/07/backing-up-azure-cosmos-db/ ale nie wiem jak to ożenić z taskiem w devops.

**Comment by wi5nia** [comment]
Masz wbudowany w Cosmosa mechanizm backupu - https://docs.microsoft.com/en-us/azure/cosmos-db/online-backup-and-restore

**Comment by krukowskid** [comment]
Kopia zapasowa musi być wykonywana na żądanie, z możliwością natychmiastowego przywrócenia. To co oferuje azure, niestety nie spełnia moich wymagań

**Reply by wi5nia** [reply]
To w linku, ktory Ci podalem masz tez sekcje jak MS rekomenduje to robic

**Reply by MichalJankowskii** [reply]
Jeśli wspomniane przez @wi5nia rozwiązanie nie jest wystrarczające, to możesz wykorzystać [`Azure Cosmos DB Data Migration Tool`](https://github.com/azure/azure-documentdb-datamigrationtool).

Wystarczy, że dodasz do DevOps i uruchomisz z command line.
